docs/2.8.0/docs/daml-finance/tutorials/advanced-topics/instrument-modeling/date-utilities.rst

```diff
@@ -0,0 +1,279 @@
+  :language: daml
+  :start-after: -- CREATE_SCHEDULE_WITH_STUB_BEGIN
+  :end-before: -- CREATE_SCHEDULE_WITH_STUB_END
+
+
+DayCount
+========
+
+Many instruments that pay interest (often expressed as an annualized rate) require an exact
+definition of how many days of interest belong to each payment period. The
+:ref:`DayCount <module-daml-finance-util-date-daycount-38239>` module provides functions for this.
+In particular, a *day count fraction* (or *dcf* in short) can be calculated between two dates. This
+indicates the fraction of a full year.
+
+For example, consider a one year bond that pays a quarterly coupon according to a given schedule,
+like the one described in the previous section.
+
+This bond will not have the same number of days in each period, which will normally result in
+different coupon amounts for each period. There are various market conventions for this, several of
+which are supported using the
+:ref:`DayCountConventionEnum <type-daml-finance-interface-types-date-daycount-daycountconventionenum-67281>`.
+
+For example, consider the schedule with a short initial stub in the previous section. Assume we want
+to use the ``Act360`` day count convention:
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/DayCount.daml
+  :language: daml
+  :start-after: -- CALCULATE_ACT360_DCF_PERIOD1_BEGIN
+  :end-before: -- CALCULATE_ACT360_DCF_PERIOD1_END
+
+which corresponds to the 2M short initial stub period. This day count fraction can be used to
+calculate a bond coupon or a swap payment for the given period, by multiplying it with the yearly
+interest rate of the instrument, e.g. 4% per annum.
+
+We can compare this initial day count fraction with those of the 2nd and 3rd periods of this
+schedule, which are both regular (3M):
+
+.. literalinclude:: ../../../src/test/daml/Daml/Finance/Util/Test/Date/DayCount.daml
+  :language: daml
+  :start-after: -- CALCULATE_ACT360_DCF_PERIOD2AND3_BEGIN
+  :end-before: -- CALCULATE_ACT360_DCF_PERIOD2AND3_END
+
+These day count fractions are clearly greater than the one of the first period, since they
+correspond to 3M instead of 2M. Note that they differ slightly, because they do not contain the
+exact same number of days.
+
+In addition to ``Act360``, the
+:ref:`DayCountConventionEnum <type-daml-finance-interface-types-date-daycount-daycountconventionenum-67281>`
+also supports several other day count conventions. Some of them can be computed using
+:ref:`calcDcf <function-daml-finance-util-date-daycount-calcdcf-20432>`, using only a start and an
+end date as an input, as described above.
+Others, e.g. ``ActActISDA``, are a bit more complicated because they require additional information.
+They can be calculated using the
+:ref:`calcPeriodDcf <function-daml-finance-util-date-daycount-calcperioddcf-63067>` function
+instead, which takes a schedule period (containing stub information) as input, as well as the
+schedule end date and the payment frequency:
```

Comment:
```suggestion
Others, such as ``ActActISDA``, are a bit more complicated because they require additional information.
You can calculate them using the
:ref:`calcPeriodDcf <function-daml-finance-util-date-daycount-calcperioddcf-63067>` function
instead. This function takes a schedule period (containing stub information) as input, as well as the
schedule end date and the payment frequency:
```

Reply:
I'd suggest adding some sort of conclusion or next steps. The page ends sort of abruptly.